### DISCUSSION TITLE
Which merge type should we use?

### DISCUSSION THREAD

**Discussion by diazona** [pytest-dev/pytest-localserver#7] [General]
@redtoad @coordt Do you have any thoughts about whether we should enforce a linear history, or force rebasing before merging but still use merge commits, or just allow merging MRs with or without rebasing first? Personally I'm a fan of the latter - if nothing else it's the most flexible - but that's just me.

Also, hope you don't mind that I enabled discussions. I figured this might be useful.

**Comment by redtoad** [comment]
I quite like "rebasing before merging but still use merge commits" which makes it easier to see what went in.

**Reply by diazona** [reply]
Cool. Though now that I look at the settings, I'm not sure if we can actually set Github to require that (which I should have checked before, my bad). I'm more used to working in GitLab, and I was thinking of its "Merge commit with semi-linear history" option.

I guess Github just allows enabling or disabling each of merge, squash, and rebase, so we could set it to only allow merges and just remember to rebase our MRs before merging. :shrug: I don't know if there's a better way.

**Reply by redtoad** [reply]
Same here. Well, let's just go for the merge then.

**Reply by diazona** [reply]
Cool, I unchecked squash and rebase. We can see what @coordt thinks too before we put this discussion to rest.

**Reply by coordt** [reply]
I'm not a huge fan of squash and rebase. I find that somehow I end up with a screwed-up rebase and lost work.

**Reply by redtoad** [reply]
Come to think of it, rebase before merge seems like too much trouble. There must be a reason why Github isn't showing the commit graphs...

**Reply by diazona** [reply]
Huh, I hadn't even noticed that. I wouldn't expect it to be messed up by rebasing, but I'm not sure.
